### PR TITLE
DOP-4592: Parse breadcrumb toc titles

### PR DESCRIPTION
**Pull request by mmeigs**
### Ticket

DOP-4592

### Notes

`slugToTitle` uses the full title fields of pages. Breadcrumbs and back/forward buttons in Snooty should use the condensed TOC labels. This creates a field `slugToBreadcrumbLabel` on the metadata as well and parses the TOC labels to have on hand if possible, and if not, uses the `title` field. 

### README updates

- - [ ] This PR introduces changes that should be reflected in the README.md and/or HACKING.md, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README.md and/or HACKING.md

### CODE REVIEW

**Comment by i80and** [comment]
Should we add a test for this in `test_postprocess.py`, e.g.

```
def test_breadcrumb_toc_titles() -> None:
    with make_test(
        {
            Path(
                "source/index.txt"
            ): """..."""
        }
    ) as result:
        assert result.metadata["slugToBreadcrumbLabel"] == <expectedValue>, "Description of this assertion"
```

**Review by i80and — APPROVED** [review]
:shipit: magnifique!

File: snooty/test_postprocess.py [hunk 1]
```diff
@@ -3383,3 +3383,60 @@ def test_orphan_diagnostic() -> None:
...
+================
+
+Alrighty
+""",
+        }
+    ) as result:
+        diagnostics = result.diagnostics[FileId("index.txt")]
+        assert len(diagnostics) == 0
+
+        slug_to_breadcrumb_label_entry = result.metadata["slugToBreadcrumbLabel"]
```
Comment (seungpark) [inline snooty/test_postprocess.py:3439]: may need to cast this to `Dict[str, str]` as this is throwing an error in the build

**Review by seungpark — APPROVED** [review]
🙌

File: snooty/test_postprocess.py (hunk 1 above)
Comment (mmeigs) [reply snooty/test_postprocess.py:3439]: THANK YOU. Had no idea what was going on.

**Pull request merged**